### PR TITLE
Update port_config.ini file parsing algorithm

### PR DESCRIPTION
**Pull request by echuawu**
### Description of PR

Update the port_config.ini file parsing algorithm to overcome some blank lines before header
Update the port_config.ini file parsing algorithm to overcome comments in the middle or at the end

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [x] 202305
- [x] 202311

#### What is the motivation for this PR?
When there are some blank lines before header line in port_config.ini, current algorithm could not handle that.
When there are comments in the middle or at the end, current algorithm could not handle that.
#### How did you do it?
Find and parse header from the end line to the upper line.
Overcome comments in the middle or at the end.
#### How did you verify/test it?
Verify it in internal regression

### CODE REVIEW

**Review by nhe-NV — APPROVED** [review]

File: ansible/roles/fanout/library/port_config_gen.py [hunk 1]
```diff
@@ -150,9 +150,9 @@ def _read_from_port_config(filepath):
         port_config = {}
         with open(filepath) as fd:
             lines = fd.readlines()
-            data_index = 0
-            while lines[data_index].startswith("#"):
-                data_index = data_index + 1
+            data_index = len(lines)
+            while not lines[data_index-1].startswith("#"):
```
Comment (roy-sror) [inline ansible/roles/fanout/library/port_config_gen.py:154]: what if there are comments at the end/middle of the file?

Comment (echuawu) [reply ansible/roles/fanout/library/port_config_gen.py]: Hi @roy-sror , originally, it was just designed to handle the blank line before header line. Anyway, your comments was reasonable. So I updated the commit to handle the blank line and comments in the middle or at the end.

**Review by bingwang-ms — APPROVED** [review]
The change looks good to me.

**Comment by bingwang-ms** [comment]
> The change looks good to me.

I will cherry-pick the change to 202311 and above. The change will not be cherry-picked to 202305 to avoid regression.

**Pull request merged**